include/zephyr/rtio/regmap.h

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2025 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_RTIO_REGMAP_H_
+#define ZEPHYR_INCLUDE_RTIO_REGMAP_H_
+
+#include <zephyr/rtio/rtio.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief A structure to describe a list of not-consecutive memory chunks
+ * for RTIO operations.
+ */
+struct rtio_reg_buf {
+	/* Valid pointer to a data buffer */
+	uint8_t *bufp;
+
+	/* Length of the buffer in bytes */
+	size_t len;
```

Comment:
If it's meant to be API then it needs to use Doxygen style comments. If not, make them "INTERNAL_HIDDEN"